### PR TITLE
wallet: Make -wallet setting not create wallets

### PR DESCRIPTION
**Pull request by ryanofsky**
This changes `-wallet` setting to only load existing wallets, not create new ones.

- Fixes settings.json corner cases reported by sjors & promag: https://github.com/bitcoin-core/gui/issues/95, https://github.com/bitcoin/bitcoin/pull/19754#issuecomment-685858578, https://github.com/bitcoin/bitcoin/pull/19754#issuecomment-685858578

- Prevents accidental creation of wallets reported most recently by jb55 http://www.erisian.com.au/bitcoin-core-dev/log-2020-09-14.html#l-355

- Simplifies behavior after #15454. #15454 took the big step of disabling creation of the default wallet. This PR extends it to avoid creating other wallets as well. With this change, new wallets just aren't created on startup, instead of sometimes being created, sometimes not. #15454 release notes are updated here and are simpler.

This change should be targeted for 0.21.0. It's a bug fix and simplifies behavior of the #15937 / #19754 / #15454 features added in 0.21.0.

---

This PR is implementing the simplest, most basic alternative listed in https://github.com/bitcoin-core/gui/issues/95#issuecomment-694236940. Other improvements mentioned there can build on top of this.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -327,15 +330,13 @@ as described below.
...
+Descriptor wallets are not the default type of wallet.
 
 In the GUI, a checkbox has been added to the Create Wallet Dialog to indicate that a
-Descriptor Wallet should be created.
+Descriptor Wallet should be created. And a `descriptors` option has been added to `createwallet` RPC.
+Setting `descriptors` to `true` will create a Descriptor Wallet instead of a Legacy Wallet.
 
-Without those options being set, a Legacy Wallet will be created instead. Additionally the
-Default Wallet created upon first startup of Bitcoin Core will be a Legacy Wallet.
+Without those options being set, a Legacy Wallet will be created instead.
```
Comment (MarcoFalke) [inline doc/release-notes.md:339]: Seems unrelated to the changes here. Probably best to put into https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21.0-Release-Notes-Draft

File: doc/release-notes.md [hunk 2]
```diff
@@ -292,15 +292,18 @@ Wallet
...
-backwards compatibility, if an unnamed `""` wallet already exists and would
-have been loaded previously, then it will still be loaded. Users without an
-unnamed `""` wallet and without any other wallets to be loaded on startup will
-be prompted to either choose a wallet to load, or to create a new wallet.
-(#15454)
+Bitcoin Core will no longer automatically create new wallets on startup. It will
+load existing wallets specified by `-wallet` options on the command line,
+wallets listed in `bitcoin.conf` or `settings.json` files, and a top-level
+unnamed ("") wallet. But if specified wallets don't exist, Bitcoin Core will log
+warnings instead of creating new wallets with random keys and addresses like
```
Comment (MarcoFalke) [inline doc/release-notes.md:301]: nit:

```suggestion
warnings instead of creating new wallets with generated keys and addresses like
```

File: src/wallet/load.cpp [hunk 3]
```diff
@@ -71,11 +71,16 @@ bool VerifyWallets(interfaces::Chain& chain)
...
         DatabaseOptions options;
         DatabaseStatus status;
+        options.require_existing = true;
         options.verify = true;
         bilingual_str error_string;
         if (!MakeWalletDatabase(wallet_file, options, status, error_string)) {
-            chain.initError(error_string);
-            return false;
+            if (status == DatabaseStatus::FAILED_NOT_FOUND) {
+                chain.initWarning(Untranslated(strprintf("Skipping -wallet path that doesn't exist. %s\n", error_string.original)));
```
Comment (MarcoFalke) [inline src/wallet/load.cpp:79]:
```suggestion
                chain.initWarning(strprintf(Untranslated("Skipping -wallet path that doesn't exist. %s\n"), error_string)));
```

File: test/functional/test_framework/test_framework.py [hunk 4]
```diff
@@ -391,9 +391,13 @@ def setup_nodes(self):
                 assert_equal(chain_info["initialblockdownload"], False)
 
     def import_deterministic_coinbase_privkeys(self):
-        wallet_names = [self.default_wallet_name] * len(self.nodes) if self.wallet_names is None else self.wallet_names
-        assert len(wallet_names) <= len(self.nodes)
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_framework.py:395]: Could keep the assert as

```
assert len(self.wallet_names) <= len(self.nodes)

**Review by maflcko — COMMENTED** [review]
left some nits. No opinion on the actual changes.

File: doc/release-notes.md (hunk 2 above)
Comment (ryanofsky) [reply doc/release-notes.md]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r507718806

> nit:

Updated

File: doc/release-notes.md (hunk 1 above)
Comment (ryanofsky) [reply doc/release-notes.md]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r507716803

> Seems unrelated to the changes here. Probably best to put into [bitcoin-core/bitcoin-devwiki/wiki/0.21.0-Release-Notes-Draft](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21.0-Release-Notes-Draft)

Thanks. Moved to wiki

File: src/wallet/load.cpp (hunk 3 above)
Comment (ryanofsky) [reply src/wallet/load.cpp:79]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r507720604

> 

It would be nice if this worked, but it doesn't currently. We only overloaded tfm::format inutil/translation.h to substitute normal arguments into bilingual format strings, not bilingual arguments into bilingual format strings

File: test/functional/test_framework/test_framework.py (hunk 4 above)
Comment (ryanofsky) [reply test/functional/test_framework/test_framework.py:395]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r507723131

> Could keep the assert as
> 
> ```
> assert len(self.wallet_names) <= len(self.nodes)
> ```

Added back

**Comment by hebasto** [comment]
Concept ACK.

**Review by ryanofsky — COMMENTED** [review]
Updated 1044521921405d1113ece4c0ebbed130bbe6ae12 -> bc2b0d99a5f54b941f8c663385f42cb9da20b94b ([`pr/nowa.1`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.1) -> [`pr/nowa.2`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/nowa.1..pr/nowa.2)) with suggested tweaks

**Review by meshcollider — COMMENTED** [review]
Concept / utACK bc2b0d99a5f54b941f8c663385f42cb9da20b94b

**Comment by hebasto** [comment]
Testing bc2b0d99a5f54b941f8c663385f42cb9da20b94b with the GUI.

Steps:
1. having two wallets: a bdb, and a sqlite
2. in the GUI open the both wallets, and shutdown
3. verifying that the both wallets are written into the `settings.json`
4. rename/remove the sqlite wallet
5. start `bitcoin-qt`, observing a nice warning (on master a new bdb wallet was created silently)
6. in the GUI only the bdb wallet is loaded _(please note this)_
7. shutdown `bitcoin-qt` and expecting to see in the `settings.json` the list of wallets that were loaded before shutdown, i.e., only the bdb wallet, but both wallets are actually listed

Is such behavior expected? If "yes", is it worth to document it?

File: doc/release-notes.md [hunk 5]
```diff
@@ -292,15 +292,18 @@ Wallet
...
-have been loaded previously, then it will still be loaded. Users without an
-unnamed `""` wallet and without any other wallets to be loaded on startup will
-be prompted to either choose a wallet to load, or to create a new wallet.
-(#15454)
+### Automatic wallet creation removed
+
+Bitcoin Core will no longer automatically create new wallets on startup. It will
+load existing wallets specified by `-wallet` options on the command line,
+wallets listed in `bitcoin.conf` or `settings.json` files, and a top-level
+unnamed ("") wallet. But if specified wallets don't exist, Bitcoin Core will log
```
Comment (hebasto) [inline doc/release-notes.md:300]: bc2b0d99a5f54b941f8c663385f42cb9da20b94b

I've read and then re-read this sentence _"It will load existing wallets specified by `-wallet` options on the command line, wallets listed in `bitcoin.conf` or `settings.json` files, and a top-level unnamed ("") wallet."_

It sounds to me like an existing top-level unnamed ("") wallet is loaded unconditionally. That is not the case, right?

**Review by tuananh11091992 — APPROVED** [review]

File: src/wallet/load.cpp (hunk 3 above)
Comment (hebasto) [inline src/wallet/load.cpp:79]: bc2b0d99a5f54b941f8c663385f42cb9da20b94b

This is a user faced message. Why keep it untranslated?

**Review by hebasto — COMMENTED** [review]
ACK bc2b0d99a5f54b941f8c663385f42cb9da20b94b

File: doc/release-notes.md (hunk 5 above)
Comment (ryanofsky) [reply doc/release-notes.md:300]: > [bc2b0d9](https://github.com/bitcoin/bitcoin/commit/bc2b0d99a5f54b941f8c663385f42cb9da20b94b)
> 
> I've read and then re-read this sentence _"It will load existing wallets specified by `-wallet` options on the command line, wallets listed in `bitcoin.conf` or `settings.json` files, and a top-level unnamed ("") wallet."_
> 
> It sounds to me like an existing top-level unnamed ("") wallet is loaded unconditionally. That is not the case, right?

I am happy to change this paragraph (and please give your suggestion), but I think it is fine. I can't think of any normal sequence of operations that would result in a "" wallet not being loaded and this paragraph isn't trying to translate every condition in wallet loading code to prose, just mention wallets that can be loaded at startup.

File: src/wallet/load.cpp (hunk 3 above)
Comment (ryanofsky) [reply src/wallet/load.cpp:79]: > This is a user faced message. Why keep it untranslated?

It's a technical message not going to be shown to a typical GUI user just loading and unloading wallets, instead of manually moving files inside wallet directories or creating a bitcoin.conf and writing a list of wallets. I think the message might also be short lived (https://github.com/bitcoin-core/gui/issues/95#issuecomment-694236940) and not worth translating right away. But I don't have a strong opinion and would be happy to change. I believe similar warnings like ones from UpdateWalletSetting are also untranslated.

**Review by ryanofsky — COMMENTED** [review]
re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r508997319 

> Is such behavior expected? If "yes", is it worth to document it?

This comment describes the behavior (which is expected behavior) but doesn't describe which parts of the behavior are good or bad, or which parts of the behavior are undocumented. Maybe reading list of alternatives https://github.com/bitcoin-core/gui/issues/95#issuecomment-694236940 would clarify approach in this PR. This PR is just doing the simplest possible thing it can do to avoid creating wallets unintentionally. Other improvements like displaying temporarily unavailable wallets or updating the settings file can build on top of this PR, but would be more complicated followups.

If the problem here is just documentation, it would help to have a specific suggestion how to improve the documentation.

File: doc/release-notes.md (hunk 5 above)
Comment (hebasto) [reply doc/release-notes.md:300]: Suggesting _"It will load existing wallets specified by `-wallet` options on the command line, wallets listed in `bitcoin.conf` or `settings.json` files."_ but not having a strong opinion about that. Feel free to ignore this comment.

File: src/wallet/load.cpp (hunk 3 above)
Comment (hebasto) [reply src/wallet/load.cpp:79]: Wallets are just files, and such a warning could be caused by user's operations with wallets.

>  I believe similar warnings like ones from UpdateWalletSetting are also untranslated.

Correct. I think the reason of that is an absence of the pr :)
Sure, no need to change here.

File: doc/release-notes.md (hunk 5 above)
Comment (ryanofsky) [reply doc/release-notes.md:300]: > Suggesting _"It will load existing wallets specified by `-wallet` options on the command line, wallets listed in `bitcoin.conf` or `settings.json` files."_ but not having a strong opinion about that. Feel free to ignore this comment.

That text seems to match current text exactly, so the suggestion is to stop mentioning the top-level wallet? That seems ok because it's not essential information, but it is information that was in the previous "Default wallet" version of the paragraph, so I'd be reluctant to lose it. It's possible for a top level wallet to not be loaded, but you have to go out of your way to do that, and I don't this release note implies that it's not possible.

File: src/wallet/load.cpp (hunk 3 above)
Comment (ryanofsky) [reply src/wallet/load.cpp:79]: > Wallets are just files, and such a warning could be caused by user's operations with wallets.

Sure. If this warning turns out to occur by common file operations, my first preference would be to implement one of the other alternatives in https://github.com/bitcoin-core/gui/issues/95#issuecomment-694236940 so the warning stops occurring

File: doc/release-notes.md [hunk 6]
```diff
@@ -292,15 +292,18 @@ Wallet
...
-Bitcoin Core will no longer create an unnamed `""` wallet by default when no
-wallet is specified on the command line or in the configuration files. For
-backwards compatibility, if an unnamed `""` wallet already exists and would
-have been loaded previously, then it will still be loaded. Users without an
-unnamed `""` wallet and without any other wallets to be loaded on startup will
-be prompted to either choose a wallet to load, or to create a new wallet.
-(#15454)
+### Automatic wallet creation removed
+
+Bitcoin Core will no longer automatically create new wallets on startup. It will
```
Comment (jonatack) [inline doc/release-notes.md:297]: maybe s/new wallets/a new wallet/

File: doc/release-notes.md [hunk 7]
```diff
@@ -292,15 +292,18 @@ Wallet
...
+load existing wallets specified by `-wallet` options on the command line,
+wallets listed in `bitcoin.conf` or `settings.json` files, and a top-level
+unnamed ("") wallet. But if specified wallets don't exist, Bitcoin Core will log
+warnings instead of creating new wallets with new keys and addresses like
```
Comment (jonatack) [inline doc/release-notes.md:301]: maybe "log a warning instead of creating a new wallet"

File: doc/release-notes.md (hunk 5 above)
Comment (jonatack) [inline doc/release-notes.md:300]: maybe s/But if specified wallets/However, if the specified wallets/

File: src/wallet/init.cpp [hunk 8]
```diff
@@ -60,7 +60,7 @@ void WalletInit::AddWalletOptions(ArgsManager& argsman) const
...
-    argsman.AddArg("-wallet=<path>", "Specify wallet database path. Can be specified multiple times to load multiple wallets. Path is interpreted relative to <walletdir> if it is not absolute, and will be created if it does not exist (as a directory containing a wallet.dat file and log files). For backwards compatibility this will also accept names of existing data files in <walletdir>.)", ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::WALLET);
+    argsman.AddArg("-wallet=<path>", "Specify wallet path to load at startup. Can be specified multiple times to load multiple wallets. Path is interpreted relative to <walletdir> if it is not absolute as a directory containing wallet data and log files. For backwards compatibility this will also accept names of existing data files in <walletdir>.) This will only load existing wallets, not create new ones.", ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::WALLET);
```
Comment (jonatack) [inline src/wallet/init.cpp:63]: - dangling bracket: `s/<walletdir>.)/<walletdir>./`

- maybe "Path is to a directory containing wallet data and log files. If the path is not absolute, it is interpreted as relative to <walletdir>."

- missing comma after "compatibility"

- mixed (inconsistent) verb tenses, consider s/this will also accept/this accepts/ and "This only loads existing wallets and does not create new ones."

**Review by jonatack — COMMENTED** [review]
Concept/approach ACK. Some optional comments below.

**Comment by jonasschnelli** [comment]
Concept ACK

**Review by ryanofsky — COMMENTED** [review]
Thanks for reviews! Just updated with documentation tweaks.

---

Updated bc2b0d99a5f54b941f8c663385f42cb9da20b94b -> e368d2bb3b4d1211f01e17dbe7e1f80528b77626 ([`pr/nowa.2`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.2) -> [`pr/nowa.3`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.3), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/nowa.2..pr/nowa.3)) with suggested changes to documentation

File: test/functional/wallet_backup.py [hunk 9]
```diff
@@ -92,10 +92,10 @@ def do_one_round(self):
...
     # As above, this mirrors the original bash test.
-    def start_three(self):
-        self.start_node(0)
-        self.start_node(1)
-        self.start_node(2)
+    def start_three(self, args=()):
+        self.start_node(0, self.extra_args[0] + list(args))
+        self.start_node(1, self.extra_args[1] + list(args))
+        self.start_node(2, self.extra_args[2] + list(args))
         connect_nodes(self.nodes[0], 3)
```
Comment (MarcoFalke) [inline test/functional/wallet_backup.py:99]: needs rebase

**Comment by ryanofsky** [comment]
Rebased e368d2bb3b4d1211f01e17dbe7e1f80528b77626 -> 01476a88a6095fd3af71cb9bf1eadef920a1197b ([`pr/nowa.3`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.3) -> [`pr/nowa.4`](https://github.com/ryanofsky/bitcoin/commits/pr/nowa.4), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/nowa.3-rebase..pr/nowa.4)) due to conflict with #19967

**Comment by achow101** [comment]
ACK 01476a88a6095fd3af71cb9bf1eadef920a1197b

**Review by luke-jr — CHANGES_REQUESTED** [review]
Concept ACK

But I think it should error, not just warn (maybe with a way to suppress the error to a warning per-wallet?). And there should probably be *some* command-line option for load-or-create...

**Review by hebasto — APPROVED** [review]
re-ACK 01476a88a6095fd3af71cb9bf1eadef920a1197b

File: test/functional/wallet_multiwallet.py [hunk 10]
```diff
@@ -80,7 +81,9 @@ def wallet_file(name):
         # rename wallet.dat to make sure plain wallet file paths (as opposed to
         # directory paths) can be loaded
         # create another dummy wallet for use in testing backups later
-        self.start_node(0, ["-nowallet", "-wallet=empty", "-wallet=plain"])
+        self.start_node(0)
+        node.createwallet("empty", descriptors=False)
```
Comment (achow101) [inline test/functional/wallet_multiwallet.py:85]: When possible, can we please avoid forcing legacy wallets? Not all of these newly created wallets need to be legacy.

Comment (ryanofsky) [reply test/functional/wallet_multiwallet.py:85]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r512266002

> When possible, can we please avoid forcing legacy wallets? Not all of these newly created wallets need to be legacy.

Obviously yes, but these wallets were legacy before this PR and they are legacy after this PR. This PR isn't trying to change wallet types, and I wasn't trying to make things more difficult for myself by introducing unnecessary changes at the same time as fixing all of these tests. But I can make the change if an update is needed. I'm reviewing #18788 which updates tests to use descriptor wallets, and think that PR is another good place to take up this issue.

Comment (achow101) [reply test/functional/wallet_multiwallet.py:85]: Hmm, then `wallet_multiwallet.py --descriptors` is currently very misleading as it doesn't actually test multiwallet descriptor wallets.

Comment (ryanofsky) [reply test/functional/wallet_multiwallet.py:85]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r512293907

> Hmm, then `wallet_multiwallet.py --descriptors` is currently very misleading as it doesn't actually test multiwallet descriptor wallets.

I'm not seeing that. There are createwallet calls that don't set descriptors=False, including the one that creates the default wallet. I know I definitely made updates to this test to deal with wallet.sqlite issues that happened only with descriptor wallets, but maybe not all of these changes survived the reverted naming. Going forward other createwallet calls can be updated and I'm happy to do it here if there need to be updates here, or I think #18788 could be another logical place to switch more legacy wallets to descriptor wallets

**Comment by laanwj** [comment]
Concept ACK

**Comment by ryanofsky** [comment]
ACK roundup

meshcollider utACK https://github.com/bitcoin/bitcoin/pull/20186#pullrequestreview-512233389
hebasto ACK https://github.com/bitcoin/bitcoin/pull/20186#pullrequestreview-516615230
achow101 ACK https://github.com/bitcoin/bitcoin/pull/20186#issuecomment-713726076

jonatack concept ACK https://github.com/bitcoin/bitcoin/pull/20186#pullrequestreview-513540739
jonasschnelli concept ACK https://github.com/bitcoin/bitcoin/pull/20186#issuecomment-713501054
luke-jr concept ACK https://github.com/bitcoin/bitcoin/pull/20186#pullrequestreview-516301051

If anyone else is curious to review, the code change is very simple. PR is 90% documentation and test changes

**Review by maflcko — APPROVED** [review]
review ACK 01476a88a6095fd3af71cb9bf1eadef920a1197b 🏂

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 01476a88a6095fd3af71cb9bf1eadef920a1197b 🏂
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgBvwv/bY+eUShaDOuvAG0i9cebopzCwinm/okeSaUY8n7+aFqqXQCLxCyDUqjd
iy7tG01DaGjtqwGNfHRzSM6aOLEokQDKG3jwwMLn6FhofEsvfUhyBPrf282pjyeQ
81CdmAwM/jZv1ZEMIRZ+NTGzoRPsSmXbIF5T8SEz1D1lxtu4ZZkQu+sP9aCCPgDW
W30fi++JD0y7E3R3Qzkzgj+XjUAfE31ishJ8omXn6RYpZ905D/XQYsi8qnRcg0Zb
tE3LhCcUa2OJaN6suBWI/C9YKP2a5y7Kv2+4DVEJHDieAsjQFiGkkmqr3WLWChZF
zuP0diiTO4pKGl0LEBfGmP58yxqBWkaBtf6J9xNg/2lOJf5Hca1qYrRQ82YUBsOq
a90o5DSuuE+Su3GPa5npdJ4TKH1cHK3zPW2ntYEHFhCheV6I46Snij0oYlq/odS0
AZHwD7Wcf1Hs3adPOqSBeFT5k1gv7j7KR/d3nCh0LzfbR7/GQtA7brDVnSUVOhjg
guDVKx3B
=B5t/
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `8c05cf2e3a6f8f2bf14f360a9f14a7a7d62deab81e5fc85dad09447ac883fef6  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401088c05cf2e3a6f8f2bf14f360a9f14a7a7d62deab81e5fc85dad09447ac883fef6f01009a9f79587b7be789214aa4b10cc623908fff0101d4293ef014f06f28226ab519bd3a87608f020a2e9f4daea02042c658c40b1fa458c78414240ac321e483c48ad570c5555c30b08f1045f9acb17f0085f74ed22a59887d10083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01033105e93cd9dfc89ec2d5b392a2a07f508f02052341645170d82533003d9522ca543750d181ebc8b52622a0d9078ec34f43f6008f1045f9acb17f0088a12d1757eaa1b080083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010737c071b9bfa96fd8ee9e9b9636f1e5c08f02053edbe7599b669026765ad904ae3783401e2ceaf6b52c5fe9b6d73e3995cea3d08f020e3385c30c21adb8c8d07ff67706166a4c491017d156398a08c1c45a7391f5a1808f1045f9acb19f008cca3c5f26217b3020083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010a14163b66c0fe6992a48b6effe06d83b08f020e034c32f2fb187c9f1c360243f8aaba9a4bc4985f31d5099193e3af414c31ddf08f0202477aa8bfeffa019156a4c93fe8fa9bc78ab6100f0fe9c857b414b9d4b2ac07d08f1045f9acb17f00896089ca3c5175b2e0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

File: doc/release-notes.md [hunk 11]
```diff
@@ -292,15 +292,18 @@ Wallet
...
+Bitcoin Core will no longer automatically create new wallets on startup. It will
+load existing wallets specified by `-wallet` options on the command line or in
+`bitcoin.conf` or `settings.json` files. And by default it will also load a
+top-level unnamed ("") wallet. However, if specified wallets don't exist,
+Bitcoin Core will now just log warnings instead of creating new wallets with
+new keys and addresses like previous releases did.
+
+New wallets can be created through the GUI (which has a more prominent create
+wallet option), through the `bitcoin-cli createwallet` or `bitcoin-wallet
+create` commands, or the `createwallet` RPC. (#15454)
```
[the remark this answers is not in the document]
Comment (ryanofsky) [reply doc/release-notes.md:306]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r514265254

Thanks, added this to https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21.0-Release-Notes-Draft

File: doc/release-notes.md [hunk 12]
```diff
@@ -292,15 +292,18 @@ Wallet
...
-have been loaded previously, then it will still be loaded. Users without an
-unnamed `""` wallet and without any other wallets to be loaded on startup will
-be prompted to either choose a wallet to load, or to create a new wallet.
-(#15454)
+### Automatic wallet creation removed
+
+Bitcoin Core will no longer automatically create new wallets on startup. It will
+load existing wallets specified by `-wallet` options on the command line or in
+`bitcoin.conf` or `settings.json` files. And by default it will also load a
+top-level unnamed ("") wallet. However, if specified wallets don't exist,
```
[the remark this answers is not in the document]
Comment (ryanofsky) [reply doc/release-notes.md:300]: re: https://github.com/bitcoin/bitcoin/pull/20186#discussion_r514276864

My brain might just be poisoned from looking at this paragraph too many times, but "if it exists" seems like a suspicious condition to add. Like going to a bank teller and saying "I am not robbing this bank today." The paragraph starts off by saying new wallets won't be created, and if a wallet doesn't exist it can't be loaded. Anyway this can be edited in the wiki if someone has a better sense

**Review by promag — COMMENTED** [review]
Code review ACK 01476a88a6.

@ryanofsky not sure if the following should be fixed:

Running with an empty datadir:

```sh
bitcoind -regtest -wallet=""
```
Gives
```
<datadir>/regtest/wallets'. Data is not in recognized format.
```
